module/applications/activity/attack-sheet.mjs

```diff
@@ -42,13 +42,29 @@ export default class AttackSheet extends ActivitySheet {
   async _prepareEffectContext(context) {
     context = await super._prepareEffectContext(context);
 
-    // TODO: Add better default label to indicate what ability will be chosen
-    // TODO: Add spellcasting option to automatically select spellcasting ability
+    const availableAbilities = this.activity.availableAbilities;
     context.abilityOptions = [
-      { value: "", label: "" },
-      ...Object.entries(CONFIG.DND5E.abilities).map(([value, config]) => ({ value, label: config.label }))
+      {
+        value: "", label: game.i18n.format("DND5E.DefaultSpecific", {
+          default: this.activity.type.classification === "spell"
+            ? game.i18n.localize("DND5E.Spellcasting").toLowerCase()
+            : availableAbilities.size
+              ? game.i18n.getListFormatter({ style: "short", type: "disjunction" }).format(
+                Array.from(availableAbilities).map(a => CONFIG.DND5E.abilities[a].label.toLowerCase())
+              )
+              : game.i18n.localize("DND5E.None").toLowerCase()
+        })
+      },
+      // TODO: Stick an <hr> here when possible
+      { value: "none", label: game.i18n.localize("DND5E.None") },
+      { value: "spellcasting", label: game.i18n.localize("DND5E.Spellcasting") },
+      ...Object.entries(CONFIG.DND5E.abilities).map(([value, config]) => ({
+        value, label: config.label, group: game.i18n.localize("DND5E.Abilities")
+      }))
     ];
 
+    context.hasBaseDamage = this.item.system.damage?.base;
```

Comment:
```suggestion
    context.hasBaseDamage = this.item.system.damage?.base.formula;
```

Reply:
I was thinking we'd always want the "Include Base Damage" checkbox to be shown on weapons, even if the user hasn't set up any damage on the weapon yet (it won't show up in the weapon parts list unless an actual formula is configured).

Reply:
Ah ok, makes sense